### DISCUSSION TITLE
Neuron is available through Whalebrew

### DISCUSSION THREAD

**Discussion by colindean** [srid/neuron#652] [Ideas]
Neuron has been added to Whalebrew's core packages. It may take a couple of minutes/hours to build and push to Docker Hub.

    whalebrew install whalebrew/neuron

I'll probably add this to the guide myself in a PR next week if someone else doesn't get to it before me.

**Comment by srid** [comment]
Nice; does it use the official [sridca/neuron](https://hub.docker.com/r/sridca/neuron) image? If not, how is the image built?

**Comment by colindean** [comment]
It does! The [`neuron` whalebrew container](https://github.com/whalebrew/whalebrew-packages/blob/master/neuron/Dockerfile) adds only port mappings, volume mapping, and a working directory.

There are some caveats to some `neuron` commands: https://github.com/whalebrew/whalebrew-packages/blob/master/neuron/README.md

**Comment by srid** [comment]
@colindean Nice. When does the whalebrew image get rebuilt? Does it rebuild on every update to the `sridca/neuron` upstream image?

**Comment by colindean** [comment]
I'm not sure of the frequency of rebuild. I'll try to find out.

**Comment by pjeorge** [comment]
I just installed neuron through `whalebrew` but was unable to reach the webserver from http://127.0.0.1:8080
```
neuron gen -wS
? Serving /Users/john/.neuron/output at http://127.0.0.1:8080

59931b3b7307   whalebrew/neuron   "neuron gen -wS"         4 minutes ago   Up 4 minutes   0.0.0.0:1440->1440/tcp, 0.0.0.0:8080->8080/tcp   hardcore_rosalind

curl: (52) Empty reply from server
```

**Comment by colindean** [comment]
Try this suggestion from the neuron whalebrew README:

> Use -s 0.0.0.0:8080 or :1440 when invoking neuron rib when you want to
use Rib's built-in server from within the container.
